From bd6b423bb2fd5dd710017cf815ed7f9549cc6ac4 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Thu, 8 Jul 2021 19:12:26 +0200
Subject: [PATCH 3/5] fdts: stm32mp15_bl2: add the trusted boot configuration

Add the trusted boot firmware configuration information into
the BL2 device tree. It enabled the authentication.
MBEDTLS heap is not defined, it will use the hard coded one.

Upstream-Status: Pending

Signed-off-by: Lionel Debieve <lionel.debieve@st.com>
Signed-off-by: Nicolas Toromanoff <nicolas.toromanoff@st.com>
Change-Id: I47181d179b814dc512dc527ae54a5a19a126f1c5
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/184833
Reviewed-by: CITOOLS <smet-aci-reviews@lists.codex.cro.st.com>
Reviewed-by: CIBUILD <smet-aci-builds@lists.codex.cro.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@st.com>
(cherry picked from commit 9b29fb46f21abead3ab96f823803d8846d82597c)
---
 fdts/stm32mp15-bl2.dtsi | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/fdts/stm32mp15-bl2.dtsi b/fdts/stm32mp15-bl2.dtsi
index d00e35b29..240952a86 100644
--- a/fdts/stm32mp15-bl2.dtsi
+++ b/fdts/stm32mp15-bl2.dtsi
@@ -86,7 +86,33 @@
 			hw_cfg_uuid = "08b8f1d9-c9cf-9349-a962-6fbc6b7265cc";
 			tos_fw_cfg_uuid = "26257c1a-dbc6-7f47-8d96-c4c4b0248021";
 			nt_fw_cfg_uuid = "28da9815-93e8-7e44-ac66-1aaf801550f9";
+#if TRUSTED_BOARD_BOOT
+			t_key_cert_uuid = "827ee890-f860-e411-a1b4-777a21b4f94c";
+			t_boot_fw_cert_uuid = "d6e269ea-5d63-e411-8d8c-9fbabe9956a5";
+			tos_fw_key_cert_uuid = "9477d603-fb60-e411-85dd-b7105b8cee04";
+			nt_fw_key_cert_uuid = "8ad5832a-fb60-e411-8aaf-df30bbc49859";
+			tos_fw_content_cert_uuid = "a49f4411-5e63-e411-8728-3f05722af33d";
+			nt_fw_content_cert_uuid = "8ec4c1f3-5d63-e411-a7a9-87ee40b23fa7";
+#endif
 		};
 	};
+
+#if TRUSTED_BOARD_BOOT
+	tb_fw-config {
+		compatible = "arm,tb_fw";
+
+		/* Disable authentication for development */
+		disable_auth = <0x0>;
+
+		/*
+		 * The following two entries are placeholders for Mbed TLS
+		 * heap information.
+		 */
+		mbedtls_heap_addr = <0x0 0x0>;
+		mbedtls_heap_size = <0x0>;
+	};
+
+#include "cot_descriptors.dtsi"
+#endif
 #endif /* !STM32MP_USE_STM32IMAGE */
 };
-- 
2.34.1

